lib/ex_doc/formatter/html.ex

```diff
@@ -375,14 +375,22 @@ defmodule ExDoc.Formatter.HTML do
   end
 
   defp generate_redirect(filename, config, redirect_to) do
-    unless File.regular?("#{config.output}/#{redirect_to}") do
+    unless module_file_exists("#{config.output}/#{redirect_to}") do
       IO.puts(:stderr, "warning: #{filename} redirects to #{redirect_to}, which does not exist")
     end
 
     content = Templates.redirect_template(config, redirect_to)
     File.write!("#{config.output}/#{filename}", content)
   end
 
+  defp module_file_exists(path) do
+    files = path |> Path.dirname() |> File.ls!()
```

Comment:
The directory may not exist, which means the `ls!` operation here can fail. Maybe check for `File.regular?` first and only if it exists, we check for the casing?

Reply:
Good point, this was based on the idea that files are generated without subdirectories but I didn't think about the case where a user just puts an invalid output directory.